store/postgres/src/deployment_store.rs

```diff
@@ -735,6 +738,25 @@ impl DeploymentStore {
         .await
     }
 
+    pub(crate) fn load_indexes(&self, site: Arc<Site>) -> Result<IndexList, StoreError> {
+        let store = self.clone();
+        let mut binding = self.get_conn()?;
+        let conn = binding.deref_mut();
+        let mut list = IndexList {
+            indexes: HashMap::new(),
+        };
+        let schema_name = site.namespace.clone();
+        let layout = store.layout(conn, site)?;
+        for (_, table) in &layout.tables {
+            let table_name = table.name.as_str();
+            let indexes = catalog::indexes_for_table(conn, schema_name.as_str(), table_name)
+                .map_err(StoreError::from)?;
+            let collect: Vec<CreateIndex> = indexes.into_iter().map(CreateIndex::parse).collect();
+            list.indexes.insert(table_name.to_string(), collect);
+        }
```

Comment:
It would probably be nicer to move the code from line 745 to here into a `IndexList::load(conn: &mut PgConnection, layout: &Layout) -> Result<Self, StoreError>` to encapsulate this a little more.

Reply:
Great suggestion!